compiler/ballerina-compiler-api/src/main/java/org/ballerina/compiler/impl/BallerinaModuleID.java

```diff
@@ -66,4 +66,24 @@ public String toString() {
         }
         return this.orgName() + "/" + this.moduleName() + ":" + this.version();
     }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+
+        if (obj == null || getClass() != obj.getClass()) {
+            return false;
+        }
+
+        // TODO: Improve this
```

Comment:
Shall we mention what needs to be improved? otherwise we might forget these over time

Reply:
Will fix these in the PR itself